### PR TITLE
Query: Add TPCTableSplittingTests

### LINKED ISSUE #27958: Add TPCTableSplittingTests

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #27958

### CODE REVIEW

File: test/EFCore.Relational.Specification.Tests/TPCTableSplittingTestBase.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        modelBuilder.Entity<Vehicle>().UseTpcMappingStrategy();
+        modelBuilder.Entity<Vehicle>().ToTable("Vehicles");
+        modelBuilder.Entity<PoweredVehicle>().ToTable("PoweredVehicles");
+        modelBuilder.Entity<CompositeVehicle>().ToTable("CompositeVehicles");
+
+        modelBuilder.Entity<Operator>(
+            eb =>
+            {
+                eb.ToTable("Vehicles");
```
Comment (smitpatel) [inline test/EFCore.Relational.Specification.Tests/TPCTableSplittingTestBase.cs:36]: @AndriySvyryd - Operator has derived type LicensedOperator and also table sharing with OperatorDetails. So couldn't use TPC here. Is that fine?

File: test/EFCore.Relational.Specification.Tests/TPCTableSplittingTestBase.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    protected override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
+        => base.AddOptions(builder).ConfigureWarnings(
+            w => w.Log(RelationalEventId.ForeignKeyTpcPrincipalWarning));
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Vehicle>().UseTpcMappingStrategy();
+        modelBuilder.Entity<Vehicle>().ToTable("Vehicles");
+        modelBuilder.Entity<PoweredVehicle>().ToTable("PoweredVehicles");
```
Comment (smitpatel) [inline test/EFCore.Relational.Specification.Tests/TPCTableSplittingTestBase.cs:30]:
```
System.InvalidOperationException : The entity type 'PoweredVehicle' is not mapped to the store object 'Vehicles'.

  Stack Trace: 
TableBase.CheckMappedEntityType(IEntityType entityType) line 134
ITableBase.GetRowInternalForeignKeys(IEntityType entityType) line 168
SharedTableEntryMap`1.GetMainEntry(IUpdateEntry entry) line 75
SharedTableEntryMap`1.GetMainEntry(IUpdateEntry entry) line 81
SharedTableEntryMap`1.GetOrAddValue(IUpdateEntry entry, SharedTableEntryValueFactory`1 createElement) line 51
CommandBatchPreparer.CreateModificationCommands(IList`1 entries, IUpdateAdapter updateAdapter, Func`1 generateParameterName) line 199
CommandBatchPreparer.BatchCommands(IList`1 entries, IUpdateAdapter updateAdapter)+MoveNext() line 67
BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection) line 57
RelationalDatabase.SaveChanges(IList`1 entries) line 57
StateManager.SaveChanges(IList`1 entriesToSave) line 1101
 ```
@AndriySvyryd - how to resolve this? Or is it a bug?

**Comment by smitpatel** [comment]
> TPC is not allowed with table-sharing

Should I add test with validation rather than query then? Can reuse the model from here but write tests from scratch.

**Comment by AndriySvyryd** [comment]
> Should I add test with validation rather than query then? Can reuse the model from here but write tests from scratch.

There should be a validation test already. You could add a query test where the principal is a leaf on TPC

**Pull request closed without merging**